pkg/bootstrap/render.go

```diff
@@ -319,6 +337,39 @@ func (b *KlusterletManifestsConfig) Generate(ctx context.Context, clientHolder *
 		renderConfig.ImagePullSecretConfig = imagePullSecretConfig
 	}
 
+	// MultipleHubs
+	// Doesn't affect on the local-cluster.
+	// Using MultipleHubs can controls the bootstrap kubeconfig secret/secrets easier.
+	multipleHubsEnabled := false
+	bootstrapKubeConfigSecrets := []BootstrapKubeConfigSecret{}
+	if local, ok := b.ManagedClusterLabels[constants.SelfManagedLabel]; !ok || local != "true" {
+		if b.klusterletconfig != nil &&
+			b.klusterletconfig.Spec.BootstrapKubeConfigs.Type == operatorv1.LocalSecrets {
+			registrationConfiguration.FeatureGates = append(registrationConfiguration.FeatureGates,
+				operatorv1.FeatureGate{
+					Feature: string(apifeature.MultipleHubs),
+					Mode:    operatorv1.FeatureGateModeTypeEnable,
+				})
+			registrationConfiguration.BootstrapKubeConfigs = b.klusterletconfig.Spec.BootstrapKubeConfigs
+
+			multipleHubsEnabled = true
+			for _, s := range b.klusterletconfig.Spec.BootstrapKubeConfigs.LocalSecrets.KubeConfigSecrets {
+				ns := os.Getenv(constants.PodNamespaceEnvVarName)
+				secret, err := clientHolder.KubeClient.CoreV1().Secrets(ns).Get(ctx, s.Name, metav1.GetOptions{})
+				if err != nil {
+					return nil, err
+				}
+				bootstrapKubeConfigSecrets = append(bootstrapKubeConfigSecrets, BootstrapKubeConfigSecret{
+					Name:       s.Name,
+					KubeConfig: base64.StdEncoding.EncodeToString(secret.Data["kubeconfig"]),
```

Comment:
should we check whether the "kubeconfig" key exists in the secret data?

Reply:
Sure, the check added.